packages/contracts-bedrock/test/L1/OPPrestateUpdater.t.sol

```diff
@@ -0,0 +1,268 @@
+
+        opcm = IOPContractsManager(
+            DeployUtils.createDeterministic({
+                _name: "OPContractsManager",
+                _args: DeployUtils.encodeConstructor(
+                    abi.encodeCall(
+                        IOPContractsManager.__constructor__,
+                        (superchainConfigProxy, protocolVersionsProxy, "dev", blueprints, impls, address(this))
+                    )
+                ),
+                _salt: DeployUtils.DEFAULT_SALT
+            })
+        );
+
+        chainDeployOutput = opcm.deploy(
+            IOPContractsManager.DeployInput({
+                roles: IOPContractsManager.Roles({
+                    opChainProxyAdminOwner: address(this),
+                    systemConfigOwner: address(this),
+                    batcher: address(this),
+                    unsafeBlockSigner: address(this),
+                    proposer: address(this),
+                    challenger: address(this)
+                }),
+                basefeeScalar: 1,
+                blobBasefeeScalar: 1,
+                startingAnchorRoot: abi.encode(
+                    OutputRoot({
+                        root: Hash.wrap(0xdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef),
+                        l2BlockNumber: 0
+                    })
+                ),
+                l2ChainId: 100,
+                saltMixer: "hello",
+                gasLimit: 30_000_000,
+                disputeGameType: GameType.wrap(1),
+                disputeAbsolutePrestate: Claim.wrap(
+                    bytes32(hex"038512e02c4c3f7bdaec27d00edf55b7155e0905301e1a88083e4e0a6764d54c")
+                ),
+                disputeMaxGameDepth: 73,
+                disputeSplitDepth: 30,
+                disputeClockExtension: Duration.wrap(10800),
+                disputeMaxClockDuration: Duration.wrap(302400)
+            })
+        );
+
+        // Also add a permissionless game
+        IOPContractsManager.AddGameInput memory input = newGameInputFactory({ permissioned: false });
+        input.disputeGameType = GameTypes.CANNON;
+        addGameType(input);
+
+        prestateUpdater = OPPrestateUpdater(
+            DeployUtils.createDeterministic({
+                _name: "OPPrestateUpdater",
+                _args: DeployUtils.encodeConstructor(
+                    abi.encodeCall(
+                        IOPContractsManager.__constructor__,
+                        (
+                            ISuperchainConfig(address(this)),
+                            IProtocolVersions(address(this)),
+                            "dev",
+                            blueprints,
+                            impls,
+                            address(0)
+                        )
+                    )
+                ),
+                _salt: DeployUtils.DEFAULT_SALT
+            })
+        );
+    }
+
+    function test_updatePrestate_succeeds() public {
```

Comment:
even if they aren't written yet, it will speed up the process if you could add empty test functions for other cases that we'll add here. 

Ie. just:

```
function test_updatePrestate_emptyPrestate_fails(){}
```

that way we can make sure we're on the same page about what tests are needed.

Reply:
Added my tests, let's talk about the ones that you think are missing on the meeting